model/task/task.go

```diff
@@ -1988,6 +1988,43 @@ func GetTimeSpent(tasks []Task) (time.Duration, time.Duration) {
 	return timeTaken, latestFinishTime.Sub(earliestStartTime)
 }
 
+// GetTaskResultsByVersion gets all tasks for a specific version
+// Query results can be filtered by task name, variant name and status in addition to being paginated and limited
+func GetTaskResultsByVersion(versionID, taskOrVariantName, sortBy string, statuses []string, sortDir, page, limit int) ([]Task, error) {
+	tasks := []Task{}
+	match := bson.M{
+		VersionKey: versionID,
+	}
+	if len(statuses) > 0 {
+		for _, status := range statuses {
+			match[StatusKey] = status
```

Comment:
This loop will only account for the last status in the statuses array. I think we should match using an "or" operator similar to what's being done with taskOrVariantName

Reply:
good catch. adding a test for this too.